c/parallel/test/test_for.cpp

```diff
@@ -52,7 +126,12 @@ C2H_TEST("for works with integral types", "[for]", integral_types)
   std::vector<T> input(num_items, T(1));
   pointer_t<T> input_ptr(input);
 
-  for_each(input_ptr, num_items, op);
+  auto& build_cache = fixture<for_build_cache_t, DeviceFor_Pointer_Fixture_Tag>::get_or_create().get_value();
+
+  std::string key_string = KeyBuilder::type_as_key<T>();
+  std::optional<std::string> test_key{key_string};
+
+  for_each(input_ptr, num_items, op, build_cache, test_key);
```

Comment:
Could this be extracted to a separate function? Since it is repeated often below?